EIPS/eip-7484.md

```diff
@@ -0,0 +1,217 @@
+---
+eip: 7484
+title: Registry Adapters for Smart Accounts
+description: Adapters that allow modular smart contract accounts to verify the security of modules using a Module Registry
+author: Konrad Kopp (@kopy-kat), zeroknots (@zeroknots)
+discussions-to: https://ethereum-magicians.org/t/erc-7484-registry-adapters-for-smart-accounts/15434
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-08-14
+requires: 4337
+---
+
+## Abstract
+
+This proposal standardises a Registry Adapter for modular smart contract accounts. This Adapter allows the account to query and verify security attestations about a module through an Attestation Registry. The adapter is responsible for querying the registry and correctly handling the return values.
+
+## Motivation
+
+[ERC-4337](./eip-4337.md) standardises the execution flow of contract accounts and [ERC-6900](./eip-6900.md) aims to standardise the modular implementation of these accounts, allowing any developer to build modules for these modular accounts (hereafter smart accounts). However, adding third-party modules into smart accounts unchecked opens up a wide range of attack vectors on these accounts.
+
+A proposed solution to these security considerations is a permission framework that allows module developers to define the permissions that their modules require. The counterpiece of these permissions are attestations that assert statements about the security of modules and can be queried onchain.
+
+This proposal is independent of the exact implementation of the permissioning system and could even be used across different kinds of permissioning systems. Instead, the goal of this proposal is to outline a standard, but flexible, way to create onchain attestations and standardise how these attestations are queried by the smart account during module installation or execution.
+
```

Comment:
I'd like to see, in your Motivation, some justification for why this needs an EIP. Generally EIPs exist to coordinate multiple vendors that have to integrate, but this looks like it'll be a single registry contract that everyone integrates with.

Reply:
Implemented in newest version